cpp/src/Slice/Parser.cpp

```diff
@@ -4242,20 +4225,12 @@ Slice::Enumerator::value() const
     return _value;
 }
 
-Slice::Enumerator::Enumerator(const ContainerPtr& container, const string& name)
-    : SyntaxTreeBase(container->unit()),
-      Contained(container, name),
-      _explicitValue(false),
-      _value(-1)
-{
-}
-
-Slice::Enumerator::Enumerator(const ContainerPtr& container, const string& name, int value)
+Slice::Enumerator::Enumerator(const ContainerPtr& container, const string& name, optional<int> value)
     : SyntaxTreeBase(container->unit()),
       Contained(container, name),
-      _explicitValue(true),
-      _value(value)
+      _explicitValue(value)
```

Comment:
I would find _hasExplicitValue much clearer.

Here, I have the impression you store value in _explicitValue.

Reply:
Or maybe we should fix the API (in a follow-up PR) with simply:

```cpp
optional<int> Slice::Enumerator::value() const { return _value; }
```

where value is an `optional<int>`.

Reply:
I renamed it to `_hasExplicitValue`!

But your proposed API isn't quite correct.
Every enumerator _must_ have a value, the code I changed is for how the parser signals:
"hey this enumerator needs an _implicit_ value calculated for it."

In a separate part of the code we check if `_value == -1`, and if so, assign it a value of `last enumerator's value + 1`